src/metabase/api/setup.clj

```diff
@@ -56,6 +58,13 @@
       ;; return user ID, session ID, and the Session object itself
       {:session-id session-id, :user-id user-id, :session session})))
 
+(defn- setup-maybe-create-and-invite-user! [{:keys [email first_name last_name] :as user}, invitor]
+  (when email
+    (if-not (email/email-configured?)
+      (log/error (trs "Could not invite user because email is not configured."))
+      (u/prog1 (user/create-and-invite-user! user invitor true)
+        (user/set-permissions-groups! <> [(group/all-users) (group/admin)])))))
```

Comment:
this line seems really heavy: `set-permissions-groups all-users admin`. It seems we have user and invitor, can we use that to set those permissions? I know we have some predicates that assume the entire userbase is user and inviter, but if this function gets reused somewhere it has enormous implications.

Reply:
> It seems we have user and invitor, can we use that to set those permissions?

Could you clarify what you mean by this?

I'm following similar logic here as how it's done in the `POST /api/user` endpoint for creating new users normally. (First a user is created, then permission groups are added).

Would it help if I renamed this `setup-maybe-create-and-invite-admin!` so it's clear that the function adds the user to the admin group?

Reply:
Ah i was misreading that line. I forgot `group/all-users` is a group. I was thinking this was setting the permissions of every user on the instance to be an admin with the assumption that there were only two users.